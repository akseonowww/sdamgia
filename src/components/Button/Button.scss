.Button {
	display: inline-block;
	padding: 0.25rem 1.1rem 0.5rem;

	font-family: inherit;
	font-size: 1rem;

	background: #d2e2ff;
	background: linear-gradient(to bottom, #d2e2ff, #9dccff);
	border: none;
	border-radius: 6px;
	outline: none;
	box-shadow: inset -1px -1px 2px rgba(77, 77, 77, 0.3);
	cursor: pointer;

	transition-duration: 0.1s;

	&:hover {
		background: #dbe8ff;
		background: linear-gradient(to bottom, #dbe8ff, #b1d6ff);
	}

	&:active {
		background: #bfd6ff;
		background: linear-gradient(to bottom, #bfd6ff, #d2e2ff);
		box-shadow: inset 1px 1px 2px rgba(77, 77, 77, 0.4);
	}
}
