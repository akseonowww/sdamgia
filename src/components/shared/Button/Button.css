.Button {
	margin: 0 !important;
	padding: 4px 15px 6px !important;

	font-size: 1rem;
}

.Button_size_big {
	padding: 10px 30px 12px !important;
}
