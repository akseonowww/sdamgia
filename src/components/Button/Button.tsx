import React, { FC, ReactNode } from 'react'
import { cn } from '@bem-react/classname'
import { classnames as cx } from '@bem-react/classnames'

import './Button.scss'

interface IButtonProps {
  className?: string
  type?: 'button' | 'submit' | 'reset'
  tabIndex?: number
  handleClick?: () => void
  children: ReactNode
}

const Button: FC<IButtonProps> = ({
  className,
  type,
  tabIndex,
  handleClick,
  children,
}) => {
  const button = cn('Button')

  return (
    <button
      className={cx(button(), className)}
      type={type}
      tabIndex={tabIndex}
      onClick={handleClick}
    >
      {children}
    </button>
  )
}

export default Button
