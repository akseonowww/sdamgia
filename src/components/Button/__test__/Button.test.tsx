import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import Button from '../Button'

describe('Button', () => {
  const handleClick = jest.fn()
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with required and optional props', () => {
    act(() => {
      render(<Button>Open</Button>, container)
    })
    expect(container && container.textContent).toBe('Open')

    act(() => {
      render(
        <Button
          className="ConstructorForm-SubmitButton"
          type="submit"
          tabIndex={0}
          handleClick={handleClick}
        >
          Make work from 5 tasks
        </Button>,
        container
      )
    })
    expect(container && container.textContent).toBe('Make work from 5 tasks')

    act(() => {
      const button = document.querySelector('.Button')
      button && button.dispatchEvent(new MouseEvent('click', { bubbles: true }))
    })
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('matches snapshot', () => {
    act(() => {
      render(<Button>Open</Button>, container)
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <Button
          className="ConstructorForm-SubmitButton"
          type="submit"
          tabIndex={0}
          handleClick={handleClick}
        >
          Make work from 5 tasks
        </Button>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
