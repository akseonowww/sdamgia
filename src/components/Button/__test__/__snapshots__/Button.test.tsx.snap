// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `"<button class=\\"Button\\">Open</button>"`;

exports[`Button matches snapshot 2`] = `"<button class=\\"Button ConstructorForm-SubmitButton\\" type=\\"submit\\" tabindex=\\"0\\">Make work from 5 tasks</button>"`;
